### PR TITLE
SPECIES NUKING 2023: Head flags 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Continues the work I started on https://github.com/tgstation/tgstation/pull/76074
Fixes https://github.com/tgstation/tgstation/issues/73782

Implements a bunch of wrapper procs for hair style, facial hair style, etc etc the whole nine yards.
Fixes many issues, especially with gradients because I did sloppy tired code that didn't work there.
Makes head/get_limb_icon() not a disgusting mess (well, at least makes it less of a disgusting mess).

## Why It's Good For The Game

Actually makes hair and lips depend entirely on the bodypart.

## Changelog

:cl:
refactor: Head rendering code has been updated significantly... Again. Please report any issues with hair, facial hair, lipstick and such.
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
Hopefully i'll be finally able to get rid of hair information being stored on human mobs, but if not eh that's okay.

**Comment by ChungusGamer666** [comment]
this will be by far the hardest species nuking PR bros

**Comment by ChungusGamer666** [comment]
everything seems to be working as expected

**Comment by ChungusGamer666** [comment]
I tried to turn this into bodypart overlays but it was a fucking mess and made the code significantly worse, so I won't do that.

**Comment by ChungusGamer666** [comment]
I will also update the head transplant unit test

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -1568,8 +1567,8 @@
 	. = ..()
 	if(!(methods & (TOUCH|VAPOR)) || !ishuman(exposed_human) || (reac_volume < 0.5))
 		return
-	exposed_human.hair_color = "#9922ff"
-	exposed_human.facial_hair_color = "#9922ff"
+	exposed_human.set_facial_haircolor("#9922ff", update = TRUE)
+	exposed_human.set_haircolor(color, update = TRUE)
```
Comment (Time-Green) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1571]: Double update intentional?

File: code/modules/surgery/bodyparts/head_hair_and_lips.dm [hunk 2]
```diff
@@ -1,121 +1,362 @@
...
-		show_debrained = FALSE
+	if(owner)
+		if(!hair_hidden && !owner.get_organ_slot(ORGAN_SLOT_BRAIN) && !HAS_TRAIT(owner, TRAIT_NO_DEBRAIN_OVERLAY))
+			show_debrained = TRUE
+		else
+			show_debrained = FALSE
 
-	if(!owner.get_organ_slot(ORGAN_SLOT_EYES) && (head_flags & HEAD_EYEHOLES))
-		show_missing_eyes = TRUE
+		if(!owner.get_organ_slot(ORGAN_SLOT_EYES))
```
Comment (Time-Green) [inline code/modules/surgery/bodyparts/head_hair_and_lips.dm:47]: Show eyeless trait thingy check?

File: code/modules/surgery/bodyparts/head_hair_and_lips.dm [hunk 3]
```diff
@@ -1,121 +1,362 @@
...
+	var/debrain_icon_state = "debrained"
+	if(bodytype & BODYTYPE_ALIEN)
+		debrain_icon = 'icons/mob/species/alien/bodyparts.dmi'
+		debrain_icon_state = "debrained_alien"
+	else if(bodytype & BODYTYPE_LARVA_PLACEHOLDER)
+		debrain_icon = 'icons/mob/species/alien/bodyparts.dmi'
+		debrain_icon_state = "debrained_larva"
+	else if(bodytype & BODYTYPE_GOLEM)
+		debrain_icon = 'icons/mob/species/golems.dmi'
+		debrain_icon_state = "debrained"
```
Comment (Time-Green) [inline code/modules/surgery/bodyparts/head_hair_and_lips.dm:172]: In a perfect world these would just be vars on the head. It's fine for now though

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -2061,8 +2062,8 @@
 		return
 
 	var/mob/living/carbon/human/exposed_human = exposed_mob
-	exposed_human.hair_color = pick(potential_colors)
-	exposed_human.facial_hair_color = pick(potential_colors)
+	exposed_human.set_facial_haircolor(pick(potential_colors), update = TRUE)
+	exposed_human.set_haircolor(pick(potential_colors), update = TRUE)
```
Comment (Time-Green) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:2066]: Double update intentional?

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by Time-Green** [comment]
Could you put in a changelog? It shouldn't be player facing but it's still a pretty big PR

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (ChungusGamer666) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1571]: Not intentional, will fix.

File: code/modules/surgery/bodyparts/head_hair_and_lips.dm (hunk 3 above)
Comment (ChungusGamer666) [reply code/modules/surgery/bodyparts/head_hair_and_lips.dm:172]: Soon (TM)

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 4 above)
Comment (ChungusGamer666) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:2066]: Unintentional. Will fix.

**Comment by ChungusGamer666** [comment]
> Could you put in a changelog? It shouldn't be player facing but it's still a pretty big PR

Done

**Comment by Time-Green** [comment]
Conflicts because the other thing got merged, if you can resolve them I’ll merge this one and move onto the next species PR

**Pull request merged**